checks.d/mysql.py

```diff
@@ -67,41 +276,46 @@ def check(self, instance):
         host, port, user, password, mysql_sock, defaults_file, tags, options, queries = \
             self._get_config(instance)
 
-        default_timeout = self.init_config.get('default_timeout', self.DEFAULT_TIMEOUT)
-
         if (not host or not user) and not defaults_file:
             raise Exception("Mysql host and user are needed.")
 
-        db = self._connect(host, port, mysql_sock, user, password, defaults_file)
+        db = self._connect(host, port, mysql_sock, user,
+                        password, defaults_file)
 
-        # Metadata collection
-        self._collect_metadata(db, host)
+        try:
+            # Metadata collection
+            self._collect_metadata(db, host)
 
-        # Metric collection
-        self._collect_metrics(host, db, tags, options, queries)
-        if Platform.is_linux():
-            self._collect_system_metrics(host, db, tags)
+            # Metric collection
+            self._collect_metrics(host, db, tags, options, queries)
+            if not Platform.is_windows():
+                self._collect_system_metrics(host, db, tags)
 
-        # Close connection
-        db.close()
+        except Exception as e:
+            self.log.debug("[MySQL] Error fetching MySQL metrics: {0}".format(str(e)))
```

Comment:
Do we need this log line ? If the exception is raised later on, it will be fully logged